src/LOAuth.ts

```diff
@@ -42,15 +42,29 @@ class LOAuth {
       ...this.appState,
       ...options.appState
     };
-    const encodedState = encodeURIComponent(JSON.stringify(state));
```

Comment:
I think you should consider extracting the client state setup into a separate function. As it's currently written, it looks like `state` is always initialized, but sometimes never used. I think if you extracted a helper function, the `clientState` variable could become a `const` and the decode state and `state` variables could be avoided in some cases.

Reply:
@mdlavin pushed a new commit with the initial state calculation into a helper method